Sources/Testing/Traits/Trait.swift

```diff
@@ -41,6 +41,22 @@ public protocol Trait: Sendable {
   ///
   /// By default, the value of this property is an empty array.
   var comments: [Comment] { get }
+
+  /// Make a copy of this trait that stores its source code representation.
+  ///
+  /// - Parameters:
+  ///   - sourceCode: The source code representation of the entire trait.
+  ///   - arguments: Source code representations of each argument to the trait,
+  ///     if any.
+  ///
+  /// - Returns: A modified copy of `self`.
+  ///
+  /// The default implementation of this method does nothing and returns `self`
+  /// unmodified.
+  func addingSourceCode(
```

Comment:
This is intended for use only by the macros, right? How about we double underscore-prefix it, but keep it `public`, to reinforce that and avoid potentially misleading users into thinking this is something they might want to use?

Reply:
Discussed offline. It's experimental API, since third parties could write traits that want access to their source code representations, but we can't use `@_spi` in this position reliably so we'll mark it with an underscore and document that it is subject to change.